pkg/graph/graph.go

```diff
@@ -79,6 +83,17 @@ func (d *Directed[V]) IncomingEdges(to V) []Edge[V] {
 	})
 }
 
+func (d *Directed[V]) RemoveVertex(v string) error {
+	err := d.underlying.RemoveVertex(v)
+	if err != nil && !errors.Is(err, graph.ErrVertexHasEdges) {
```

Comment:
What's the behaviour if it does return `ErrVertexHasEdges`? Does it remove it and its edges? Does it not remove it? Does it leave it in a weird state where the node is removed but not the edges?

Reply:
if it has edges it wont remove the node, so thats why for that error we should error. This is a bug though and i dont error for it, so good catch